Translators/WZDx/geospatial_service.py

```diff
@@ -0,0 +1,177 @@
+import os
+import requests
+import urllib3
+urllib3.disable_warnings()
+
+
+def get_geospatial_endpoint():
+    return os.environ['dual_carriageway_endpoint']
+
+
+def get_map_server_endpoint():
+    return os.environ['dual_carriageway_endpoint'].split('/exts', 1)[0]
+
+
+def measure_at_point(lat, lon, routeId) -> int:
+    '''Finds the measure at a given point for a given route
+
+    Parameters:
+        lat (float): Latitude of point
+        lon (float): Longitude of point
+        route (str): Name of route
+
+    Returns:
+        measure (int): Measure of point on route'''
+    r = requests.get(
+        f'{get_geospatial_endpoint()}/MeasureAtPoint?x={lon}&y={lat}&inSR=4326&routeId={routeId}&tolerance=&outSR=4326&f=pjson', verify=False)
+    data = r.json()
+    return int(data["features"][0]["attributes"]["Measure"])
+
+
+def point_at_measure(measure, routeId):
+    '''Finds the coordinate at a given measure for a given route
+
+    Parameters:
+        measure (int): Measure to translate to point 
+        route (str): Name of route
+
+    Returns:
+        point (dict): Coordinate of point on route'''
+    r = requests.get(
+        f'{get_geospatial_endpoint()}/PointAtMeasure?routeId={routeId}&measure={measure}&inSR=4326&outSR=4326&f=pjson', verify=False)
+    data = r.json()
+    if 'error' in data:
+        print(data)
+        return None
+    return {
+        "longitude": data['features'][0]['geometry']['x'],
+        "latitude": data['features'][0]['geometry']['y']
+    }
+
+
+def get_ten_meter_extent(lon, lat):
+    '''Finds the extent approximately 10 meter each direction around a given coordinate
+
+    Parameters:
+        lat (float): Latitude of point
+        lon (float): Longitude of point
+
+    Returns:
+        extent (str): A string representing the extent of the 10 meter square'''
+    maxLon = lon + 0.0001
+    minLon = lon - 0.0001
+    maxLat = lat + 0.0001
+    minLat = lat - 0.0001
+    return f'{minLon},{minLat},{maxLon},{maxLat}'
+
+
+def point_to_route_id(lon, lat):
+    '''Finds the route ID for a given coordinate
+
+    Parameters:
+        lat (float): Latitude of point
+        lon (float): Longitude of point
+
+    Returns:
+        route (str): Route ID'''
+    # colorado_extent = "-109.081667,37.002220,-102.028444,40.979583"
+    # NOTE: currently failing to find proper routes on occasion...
+    r = requests.get(
+        f'{get_map_server_endpoint()}/identify?geometry={lon},{lat}&geometryType=esriGeometryPoint&sr=4326&tolerance=50&mapExtent={get_ten_meter_extent(lon,lat)}&imageDisplay=600,550,96&returnGeometry=false&returnZ=false&returnM=false&returnUnformattedValues=false&returnFieldName=false&f=json', verify=False)
+    data = r.json()
+    if (len(data['results']) > 1):
+        str_results = [*map(lambda x: x['attributes']
+                            ['RouteId_Legacy'], data['results'])]

```

Comment:
To use the non-dual carriageway endpoint, I had to change RouteId_Legacy to ROUTE. Same 2 lines lower

Reply:
Not suggesting a change, just letting you know

Reply:
Sounds good, the "Route" value is different that the RouteId_Legacy in the dual carriageway instance and Route occasionally seems to return invalid and incorrect responses for what we want.